### PR TITLE
Handle functions not yet loaded, gracefully

### PR DESCRIPTION
**Pull request by narendraj9**
For functions that are not yet loaded (for example 'erc), the current code crashes because 
```elisp
(-map #'helpful--format-argument
                 (help-function-arglist sym))
```
tries to call `#'helpful--format-argument` on chars.

### CODE REVIEW

**Comment by narendraj9** [comment]
@Wilfred Can you tell me how you run tests locally?

**Comment by Wilfred** [comment]
@narendraj9 

```
# Install dependencies
$ cask
# Run tests with ert-runner
$ cask exec ert-runner
```

**Comment by Wilfred** [comment]
We should definitely support autoloaded functions. I'm tempted to just load the function and show the full docs. This is an improvement though 👍

File: helpful.el [hunk 1]
```diff
@@ -46,7 +46,7 @@
 (require 'dash)
 (require 's)
 (require 'find-func)
-(require 'nadvice)
+(require 'advice)
```
Comment (Wilfred) [inline helpful.el]: Why did you change this line?

File: test/unit-test.el [hunk 2]
```diff
@@ -112,3 +114,9 @@ variables defined without `defvar'."
 (ert-deftest helpful-variable ()
   "Smoke test for `helpful-variable'."
   (helpful-variable 'tab-width))
+
+(ert-deftest helfpul--signature ()
```
Comment (Wilfred) [inline test/unit-test.el]: Typo *helpful and *autoloaded.

File: helpful.el (hunk 1 above)
Comment (narendraj9) [reply helpful.el]: I saw that the builds were (are?) red on master and checked travis to bind that it complained by nadvice. I didn't know that 'nadvice exists. I will fix it there

File: test/unit-test.el (hunk 2 above)
Comment (narendraj9) [reply test/unit-test.el]: Sorry, I will fix these.

**Comment by narendraj9** [comment]
@Wilfred I have made changes you suggested. I think something is wrong with the tests already: 
https://travis-ci.org/Wilfred/helpful/jobs/280442584

For example, the error is about `nadvice' not found.

**Comment by Wilfred** [comment]
Yeah, something looks very broken. I re-ran a green build and it went red: https://travis-ci.org/Wilfred/helpful/jobs/273904437

**Comment by Wilfred** [comment]
OK, tested locally and this definitely works, so merging. Thanks for your PR! :)

I will try to figure out why the tests have started failing.
